content-src/asrouter/components/FxASignupForm/FxASignupForm.jsx

```diff
@@ -0,0 +1,174 @@
+
+import { actionCreators as ac } from "common/Actions.jsm";
+import { addUtmParams } from "../../templates/FirstRun/addUtmParams";
+import React from "react";
+
+export class FxASignupForm extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onInputInvalid = this.onInputInvalid.bind(this);
+    this.handleSignIn = this.handleSignIn.bind(this);
+
+    this.state = {
+      emailInput: "",
+    };
+  }
+
+  get email() {
+    return this.props.document
+      .getElementById("fxaSignupForm")
+      .querySelector("input[name=email]");
+  }
+
+  get signInURL() {
+    const urlObj = new URL(this.props.fxaEndpoint);
+    urlObj.pathname = "signin";
+    return urlObj.toString();
+  }
+
+  onSubmit(event) {
+    let userEvent = "SUBMIT_EMAIL";
+    const { email } = event.target.elements;
+    if (email.disabled) {
+      event.target.setAttribute("action", this.signInURL);
+      userEvent = "SUBMIT_SIGNIN";
+    } else if (!email.value.length) {
+      email.required = true;
+      email.checkValidity();
+      event.preventDefault();
+      return;
+    }
+
+    // Report to telemetry additional information about the form submission.
+    const value = { has_flow_params: !!this.props.flowParams.flowId.length };
+    this.props.dispatch(ac.UserEvent({ event: userEvent, value }));
+
+    global.addEventListener("visibilitychange", this.props.onClose);
+  }
+
+  handleSignIn(event) {
+    // Set disabled to prevent email from appearing in url resulting in the wrong page
+    this.email.disabled = true;
+  }
+
+  componentDidMount() {
+    // Start with focus in the email input box
+    if (this.email) {
+      this.email.focus();
+    }
+  }
+
+  onInputChange(e) {
+    let error = e.target.previousSibling;
+    this.setState({ emailInput: e.target.value });
+    error.classList.remove("active");
+    e.target.classList.remove("invalid");
+  }
+
+  onInputInvalid(e) {
+    let error = e.target.previousSibling;
+    error.classList.add("active");
+    e.target.classList.add("invalid");
+    e.preventDefault(); // Override built-in form validation popup
+    e.target.focus();
+  }
+
+  render() {
+    const { content, UTMTerm } = this.props;
+    return (
+      <div
+        id="fxaSignupForm"
+        role="group"
+        aria-labelledby="joinFormHeader"
+        aria-describedby="joinFormBody"
+        className="fxaSignupForm"
+      >
+        <h3 id="joinFormHeader" data-l10n-id={content.form.title.string_id} />
+        <p id="joinFormBody" data-l10n-id={content.form.text.string_id} />
+        <form
+          method="get"
+          action={this.props.fxaEndpoint}
+          target="_blank"
+          rel="noopener noreferrer"
+          onSubmit={this.onSubmit}
+        >
+          <input name="service" type="hidden" value="sync" />
+          <input name="action" type="hidden" value="email" />
```

Comment:
Just making sure, this is expected for signin as well? We do get an extra url param `user_action`:

```
https://accounts.firefox.com/signin
service=sync
action=email
context=fx_desktop_v3
entrypoint=activity-stream-firstrun
utm_source=activity-stream
utm_campaign=firstrun
utm_term=trailhead-full_page_e
device_id=bc00…
flow_id=68c4…
flow_begin_time=1570…
style=trailhead
user_action=signin
```

vs email submit:
```
https://accounts.firefox.com/
service=sync
action=email
context=fx_desktop_v3
entrypoint=activity-stream-firstrun
utm_source=activity-stream
utm_campaign=firstrun
utm_term=trailhead-full_page_e
device_id=7a14…
flow_id=12c7…
flow_begin_time=1570…
style=trailhead
email=email%40addr
```

Not sure what FxA is expecting, but maybe it might want to be action=signin for signin even though one could already differentiate with `user_action=signin` and that the url is different. ??

Reply:
Looking at signin flow from https://www.mozilla.org/en-US/firefox/accounts (need to be not already signed in)

we should remove action=email and user_action=signin something like

https://accounts.firefox.com/signin?service=sync&context=fx_desktop_v3&entrypoint=activity-stream-firstrun&utm_source=activity-stream&utm_campaign=firstrun&utm_term=trailhead-full_page_e&device_id=d98469719a11465e80601ac10294337a&flow_id=79cfabaf192e4b358de9e6899d317d9cd52cfbb817fd8152ecfd716d1382fbe0&flow_begin_time=1570734910334&style=trailhead

and have action= email only for signup flow. To make this happen we need to set action.disabled to true inside onSubmit
   ```
const { email, action } = event.target.elements;
    if (email.disabled) {
      action.disabled = true;
      event.target.setAttribute("action", this.signInURL);
      userEvent = "SUBMIT_SIGNIN";
    }
```